types/underscore/index.d.ts

```diff
@@ -3814,150 +3814,144 @@ declare module _ {
         propertyOf(object: object): (key: string | number | Array<string | number>) => any;
 
         /**
-        * Performs an optimized deep comparison between the two objects,
-        * to determine if they should be considered equal.
-        * @param object Compare to `other`.
-        * @param other Compare to `object`.
-        * @return True if `object` is equal to `other`.
-        **/
+         * Performs an optimized deep comparison between the two objects
+         * to determine if they should be considered equal.
+         * @param object Compare to `other`.
+         * @param other Compare to `object`.
+         * @returns True if `object` is equal to `other`.
+         **/
         isEqual(object: any, other: any): boolean;
 
         /**
-        * Returns true if object contains no values.
-        * @param object Check if this object has no properties or values.
-        * @return True if `object` is empty.
-        **/
+         * Returns true if an enumerable object contains no values (no enumerable own-properties).
+         * For strings and array-like objects checks if the length property is 0.
+         * @param object The object to check.
+         * @returns True if `object` is empty.
+         **/
         isEmpty(object: any): boolean;
 
         /**
-        * Returns true if the keys and values in `properties` matches with the `object` properties.
-        * @param object Object to be compared with `properties`.
-        * @param properties Properties be compared with `object`
-        * @return True if `object` has matching keys and values, otherwise false.
-        **/
+         * Tells you if the keys and values in properties are contained in object.
```

Comment:
Your sentence is more grammatical but the old sentence was a bit more precise. I suggest a combination.

```suggestion
         * Returns true if the keys and values in `properties` are contained in `object`.
```